hades/__init__.py

```diff
@@ -219,23 +220,20 @@ def submit():
         user.email += f", {request.form['email_second_person']}"
         user.department += f", {request.form['department_second_person']}"
 
-    # Ensure that no data is duplicated. If anything is wrong, display the corresponding error to the user
-    data = user.validate()
-    if data is not True:
-        return data
-
     # Generate the QRCode based on the given data and store base64 encoded version of it to email
     if 'no_qr' not in request.form:
         img = generate_qr(user)
         img.save('qr.png')
         img_data = open('qr.png', 'rb').read()
         encoded = base64.b64encode(img_data).decode()
 
-    # Add the user to the database and commit the transaction, ensuring no integrity errors.
-    success, reason = insert(user)
+    # Add the user to the database, ensuring no integrity errors.
+    success, reason = insert([user])
     if not success:
         log(f'Could not insert user {user}')
         log(reason)
+        if reason == 'not_unique':
```

Comment:
Above comment is basically to avoid such stuff, you can check what I did in `db_utils.insert()` for this